meshroom/core/node.py

```diff
@@ -689,11 +689,16 @@ def toDict(self):
 
     def _computeUids(self):
         """ Compute node uids by combining associated attributes' uids. """
-        for uidIndex, associatedAttributes in self.attributesPerUid.items():
-            # uid is computed by hashing the sorted list of tuple (name, value) of all attributes impacting this uid
-            uidAttributes = [(a.getName(), a.uid(uidIndex)) for a in associatedAttributes if a.enabled]
-            uidAttributes.sort()
-            self._uids[uidIndex] = hashValue(uidAttributes)
+        conditionalUids = []  # uids that are taken into account in the final computation only if their attribute's value is not empty; their UID group is -1
```

Comment:
I don't see the reason to mix independant notions.
It's better to just add a property as it's done for the "enabled" property.

Reply:
Maybe add `uidIgnoreValue=None`.
So for a string we can set if to `uidIgnoreValue=""` to ignore an empty string.

Reply:
Done. `uidIgnoreValue` is set to `None` by default and is only exposed in the StringParam constructor for now.